test/cri-containerd/main.go

```diff
@@ -195,3 +198,59 @@ func findTestNvidiaGPUDevice() (string, error) {
 	}
 	return strings.TrimSpace(string(out)), nil
 }
+
+// Checks to see if the user has the New-CredentialSpec cmdlet installed.
+func hasCmdlet() error {
+	cmdLet := "New-CredentialSpec"
+	input := fmt.Sprintf("Get-Command %s", cmdLet)
+	_, err := exec.Command(
+		"powershell",
+		input,
+	).Output()
+	if err != nil {
+		return fmt.Errorf("necessary cmdlet not installed. Please run 'Install-Module %s'", cmdLet)
+	}
+	return nil
+}
+
+// Generates a gmsa credential spec and places it at 'path'. Dir has to exist it will not be created by
+// New-CredentialSpec. Could extend to os.Stat and if it doesn't exist could mkdir but will just use ContainerPlat
+// dir for now.
+func generateCredSpec(path string) error {
+	cmd := fmt.Sprintf(`New-CredentialSpec -AccountName %s -Path "%s"`, testGmsaAccount, path)
+	_, err := exec.Command(
+		"powershell",
+		cmd,
+	).Output()
+	if err != nil {
+		return fmt.Errorf("the account specified does not exist")
```

Comment:
Is the case that the account does not exist the only error that could happen from this call?

Reply:
It's either the user does not have access to the account or the account just doesn't exist in the domain at all, so actually there are two! 👍 I'll fix this

Reply:
Resolved as I'm just returning the error directly as you suggested